vcell-core/src/main/java/cbit/vcell/export/server/N5Exporter.java

```diff
@@ -0,0 +1,273 @@
+		// and the only reason why this wouldn't work is due to some misconfiguration, but what is that misconfig
+
+		ArrayList<DataIdentifier> species = new ArrayList<>();
+		for (String specie: exportSpecs.getVariableSpecs().getVariableNames()){
+			species.add(getSpecificDI(specie));
+		}
+
+
+		for (DataIdentifier specie: species) {
+			if (unsupportedTypes.contains(specie.getVariableType())) {
+				throw new RuntimeException("Tried to export a variable type that is not supported!");
+			}
+		}
+	//        DexDataIdentifier.getVariableType();
+
+		int numVariables = species.size();
+		int numTimes = exportSpecs.getTimeSpecs().getEndTimeIndex() - exportSpecs.getTimeSpecs().getBeginTimeIndex();
+		long[] dimensions = {vcData.getMesh().getSizeX(), vcData.getMesh().getSizeY(), numVariables, vcData.getMesh().getSizeZ(), numTimes};
+		// 51X, 51Y, 1Z, 1C, 2T
+		int[] blockSize = {vcData.getMesh().getSizeX(), vcData.getMesh().getSizeY(), 1, vcData.getMesh().getSizeZ(), 1};
+
+
+		// rewrite so that it still results in a tmp file does not raise File already exists error
+		N5FSWriter n5FSWriter = new N5FSWriter(getN5FileAbsolutePath(), new GsonBuilder());
+		DatasetAttributes datasetAttributes = new DatasetAttributes(dimensions, blockSize, org.janelia.saalfeldlab.n5.DataType.FLOAT64, n5Specs.getCompression());
+		HashMap<String, Object> additionalMetaData = new HashMap<>();
+
+		String dataSetName = getN5DataSetTemplatedName(n5Specs.dataSetName);
+
+		n5FSWriter.createDataset(dataSetName, datasetAttributes);
+		N5Specs.imageJMetaData(n5FSWriter, dataSetName, vcData, numVariables, additionalMetaData);
+
+
+		for (int variableIndex=0; variableIndex < numVariables; variableIndex++){
+			//place to add tracking, each variable can be measured in tracking
+			double varFrac = (double) variableIndex / numVariables;
+			for (int timeIndex=exportSpecs.getTimeSpecs().getBeginTimeIndex(); timeIndex < exportSpecs.getTimeSpecs().getEndTimeIndex(); timeIndex++){
+				//another place to add tracking, each time index can be used to determine how much has been exported
+				// data does get returned, but it does not seem to cover the entire region of space, but only returns regions where there is activity
+				double[] data = this.dataSetController.getSimDataBlock(outputContext, this.vcDataID, species.get(variableIndex).getName(), allTimes[timeIndex]).getData();
+				DoubleArrayDataBlock doubleArrayDataBlock = new DoubleArrayDataBlock(blockSize, new long[]{0, 0, variableIndex, 0, timeIndex}, data);
+				n5FSWriter.writeBlock(dataSetName, datasetAttributes, doubleArrayDataBlock);
+				if(timeIndex % 5 == 0){
+					double timeFrac = (double) timeIndex / exportSpecs.getTimeSpecs().getEndTimeIndex();
+					double progress = (varFrac + timeFrac) / (numVariables + exportSpecs.getTimeSpecs().getEndTimeIndex());
+					exportServiceImpl.fireExportProgress(jobID, vcDataID, "N5", progress);
+				}

```

Comment:
consider using string literals for N5 and .n5 ...

Reply:
Changed all instances of N5 string to a literal for the N5 file suffix https://github.com/virtualcell/vcell/pull/1047/commits/22e070427f50b17d353fcf2724fc88025bfee054